server/base/src/main/java/org/apache/accumulo/server/util/ZooZap.java

```diff
@@ -98,76 +97,70 @@ public void execute(String[] args) throws Exception {
         SecurityUtil.serverLogin(siteConf);
       }
 
-      String volDir = VolumeConfiguration.getVolumeUris(siteConf).iterator().next();
-      Path instanceDir = new Path(volDir, "instance_id");
-      InstanceId iid = VolumeManager.getInstanceIDFromHdfs(instanceDir, new Configuration());
-      ZooReaderWriter zoo = new ZooReaderWriter(siteConf);
-
-      if (opts.zapManager) {
-        String managerLockPath = Constants.ZROOT + "/" + iid + Constants.ZMANAGER_LOCK;
-
-        try {
-          zapDirectory(zoo, managerLockPath, opts);
-        } catch (KeeperException | InterruptedException e) {
-          e.printStackTrace();
+      try (var context = new ServerContext(siteConf)) {
+        final ZooReaderWriter zoo = context.getZooReaderWriter();
+        if (opts.zapManager) {
+          ServiceLockPath managerLockPath = context.getServerPaths().createManagerPath();
+          try {
+            zapDirectory(zoo, managerLockPath, opts);
+          } catch (KeeperException | InterruptedException e) {
+            e.printStackTrace();
+          }
         }
-      }
 
-      if (opts.zapTservers) {
-        String tserversPath = Constants.ZROOT + "/" + iid + Constants.ZTSERVERS;
-        try {
-          List<String> children = zoo.getChildren(tserversPath);
-          for (String child : children) {
-            message("Deleting " + tserversPath + "/" + child + " from zookeeper", opts);
-
-            if (opts.zapManager) {
-              zoo.recursiveDelete(tserversPath + "/" + child, NodeMissingPolicy.SKIP);
-            } else {
-              var zLockPath = ServiceLock.path(tserversPath + "/" + child);
-              if (!zoo.getChildren(zLockPath.toString()).isEmpty()) {
-                if (!ServiceLock.deleteLock(zoo, zLockPath, "tserver")) {
-                  message("Did not delete " + tserversPath + "/" + child, opts);
+        if (opts.zapTservers) {
+          try {
+            Set<ServiceLockPath> tserverLockPaths =
```

Comment:
Not sure about this, but this code may have been cleaning up empty lock nodes.  If it was doing that, it may not be doing that now.